src/cryptonote_core/service_node_list.h

```diff
@@ -275,6 +276,8 @@ namespace service_nodes
     bool m_hooks_registered;
     block_height m_height;
 
+    crypto::public_key const *m_service_node_pubkey;
```

Comment:
I would use `boost::optional` for readability in this case, but a pointer works fine as well.